File: src/python/maps/map31.py
import cabin

def to30(engineRef):
    x = engineRef.player.x - 160
    yield from engineRef.mapSwitchTask('map30.ika-map', (6 * 16 + x, 16))
    if 'nearend' not in engineRef.saveFlags:
        yield from cabin.sceneTask(engineRef, 'nearend')

def to32(engineRef):
    # no adjustment here on purpose
    yield from engineRef.mapSwitchTask('map32.ika-map', (25 * 16, 38 * 16))

File: src/winter/maps/map31.ts
import { MapScript } from "./mapscript.js"
import { sceneTask } from "./../cabin.js"
import { PyEngine } from "./../winter.js"

export default new MapScript(autoexec, { to30, to32 })

export function autoexec(_engineRef: PyEngine) {}

export function *to30(engineRef: PyEngine) {
    const p = engineRef.getPlayerEntity()
    const x = p.sprite.x - 160
    yield* engineRef.mapSwitchTask('map30.ika-map', [6 * 16 + x, 16])
    if (!engineRef.hasSaveFlag('nearend')) {
        yield* sceneTask(engineRef, 'nearend')
    }
}

export function *to32(engineRef: PyEngine) {
    // no adjustment here on purpose
    yield* engineRef.mapSwitchTask('map32.ika-map', [25 * 16, 38 * 16])
}
